### PR TITLE
SONARCS-621 use cpd token info by SonarAnalyzer

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/ProtobufImporter.java [hunk 1]
```diff
@@ -20,15 +20,14 @@
...
-public interface ProtobufImporter {
-  /**
-   * Parse a protobuf binary file and import into the sensor context.
-   *
-   * @param context  the SensorContext to import into
-   * @param protobuf a File object of a protobuf binary file
-   */
-  void parse(SensorContext context, File protobuf);
+public interface ProtobufImporter extends BiConsumer<SensorContext, File> {
+  void accept(SensorContext context, File protobuf);
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/ProtobufImporter.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1161)

File: src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/CPDTokensImporter.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins.protobuf;
+
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.cpd.NewCpdTokens;
+import org.sonar.api.batch.sensor.symbol.NewSymbol;
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/CPDTokensImporter.java]: MINOR Remove this unused import 'org.sonar.api.batch.sensor.symbol.NewSymbol'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/CPDTokensImporter.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.sonar.api.batch.sensor.symbol.NewSymbolTable;
```
Comment (SonarTech) [inline src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/CPDTokensImporter.java]: MINOR Remove this unused import 'org.sonar.api.batch.sensor.symbol.NewSymbolTable'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/CPDTokensImporter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    super(SonarAnalyzer.CopyPasteTokenInfo.parser());
+  }
+
+  @Override
+  void importMessage(SensorContext context, SonarAnalyzer.CopyPasteTokenInfo message) {
+    InputFile inputFile = getInputFile(context.fileSystem(), message.getFilePath());
+    NewCpdTokens cpdTokens = context.newCpdTokens().onFile(inputFile);
+
+    for (SonarAnalyzer.CopyPasteTokenInfo.TokenInfo tokenInfo : message.getTokenInfoList()) {
+      cpdTokens.addToken(toTextRange(inputFile, tokenInfo.getTextRange()), tokenInfo.getTokenType());
```
Comment (henryju) [inline src/main/java/org/sonarsource/dotnet/shared/plugins/protobuf/CPDTokensImporter.java]: I missed that during review, but it should not be tokenInfo.getTokenType() but more something like tokenInfo.getTokenValue() or tokenInfo.getTokenImage() (cc @tvsonar )

**Pull request merged**